### DISCUSSION TITLE
keycloak-js forces refresh of Refreshing Token

### DISCUSSION THREAD

**Discussion by NL-Moritz** [keycloak/keycloak#15006] [Q&A]
Hey, I wonder about the behaviour of the KC js adapter when I trigger a reload of my page.

My setup: built a webapp and using KC for authentication. If user is not logged in, he will be redirected to my KC server and redirected back with an authorization code afterwards (onLoad=login-required and KC Standard Flow). The authorization code is then used to get access, refresh and id tokens on the token endpoint. So far so good. 

I store the tokens inside my local storage and pass them to the KC adapter when initializing it. But when I reload my page/webapp, a request is being made to the token endpoint, containing the refresh token and is responded with a new access, refresh and id tokens. 

After digging deeper into the code (of the KC js module), I found out that KC updates the tokens when minValidity == -1. This is always the case when loginIframe.enable == false. Since I have to pass the prop checkLoginIframe: false, I cannot prevent the request to the token endpoint. checkLoginIframe cannot be enabled because of the problems emphasized in the documentation: https://www.keycloak.org/docs/latest/securing_apps/#session-status-iframe

So, why does KC always forces a refresh of all my tokens even if they still valid? Wouldn't it be more efficient to check the validity of the tokens as it is currently done anyway?

**Comment by dhiren-singh-007** [comment]
I just started using keycloak and having same issue .
Did you find any solution or is it standard ?

**Comment by Frostbourn** [comment]
Any news on that issue?

**Comment by handasolo** [comment]
Hey @NL-Moritz, did you end up resolving this issue somehow?
